File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int> mp1, mp2;
        for(int i=0; i<arr.size(); i++) mp1[arr[i]]++;
        for(auto i: mp1){
            mp2[i.second]++;
            if(mp2[i.second]>1) return false;
        }
        return true;
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for(int i=0; i<arr.length; i++){
            mp.put(arr[i], mp.getOrDefault(arr[i], 0)+1);
        }
        HashSet<Integer> st = new HashSet<>();
        for(Map.Entry<Integer, Integer> x: mp.entrySet()){
            if(st.contains(x.getValue())) return false;
            st.add(x.getValue());
        }
        return true;
    }
}